### ISSUE TITLE
Signal handlers called without arguments

### ISSUE THREAD

**Issue by ulope** [gevent/gevent#1529]
* gevent version: 1.4.0, 1.5a3
* Env 1
   * Python version: CPython 3.7.6 (source, built with `pythonz`)
   * Operating System: macOS 10.15.2
* Env 2
   * Python version: CPython 3.7.6 (docker image `python:3.7` from hub)
   * Operating System: Debian 10 / 4.9.125-linuxkit
* Env 3
   * Python version: CPython 3.6.10 (docker image `python:3.6` from hub)
   * Operating System: Debian 10 / 4.9.125-linuxkit

### Description:

According to the docs `gevent.signal()` behaves identical to `signal.signal()`.

This appears to not be true. The signal handler is called without arguments which leads to a `TypeError`:
```
TypeError: handler() missing 2 required positional arguments: 'sig' and 'frame'
```

### What I've run:

gevent example (crashes with `TypeError`):
```python
import signal

import gevent
from gevent.event import Event

stop = Event()

def handler(sig, frame):
    print(sig, frame)
    stop.set()

def work():
    while not stop.is_set():
        gevent.sleep(.1)

gevent.signal(signal.SIGINT, handler)
gevent.signal(signal.SIGQUIT, handler)
gevent.signal(signal.SIGTERM, handler)

print("Handler installed. Press Ctrl-C.")

gevent.spawn(work)
stop.wait()
```

The same without gevent (works):
```python
import signal
import threading
import time


stop = threading.Event()

def handler(sig, frame):
    print(sig, frame)
    stop.set()

def work():
    while not stop.is_set():
        time.sleep(0.1)

signal.signal(signal.SIGINT, handler)
signal.signal(signal.SIGQUIT, handler)
signal.signal(signal.SIGTERM, handler)

print("Handler installed. Press Ctrl-C.")

threading.Thread(target=work, daemon=True).start()
stop.wait()
```

**Comment by jamadden** [comment]
> According to the docs `gevent.signal()` behaves identical to `signal.signal()`.

Can you point me to where you read that? As far as I can see, [the documentation says](http://www.gevent.org/api/gevent.html#signals) that `gevent.signal` is a deprecated alias for `gevent.signal_handler`. The function [documented to behave like](http://www.gevent.org/api/gevent.signal.html#gevent.signal.signal) `signal.signal` is `gevent.signal.signal`.

Your gevent example should use this:
```python
gevent.signal.signal(signal.SIGINT, handler)
gevent.signal.signal(signal.SIGQUIT, handler)
gevent.signal.signal(signal.SIGTERM, handler)
```

Or begin with this:
```python
from gevent.signal import signal
```

and then proceed as for the non-gevent example.

**Comment by ulope** [comment]
Indeed it does. 
I was reading http://www.gevent.org/api/gevent.signal.html (which is the first for `gevent signal` in google) and missed that it was talking about `gevent.signal.signal` instead of `gevent.signal`. 

Maybe adding a note here to avoid this confusion would be nice.

**Comment by jamadden** [comment]
The old name `gevent.signal` has been deprecated for almost 5 years. Back in 2015 in https://github.com/gevent/gevent/issues/648 a github search found a number of direct uses of `gevent.signal()` in existing code, but I'm not seeing that now. Give that it requires a number of unsavory shenanigans in the code base, *and* is confusing, I'm strongly tempted to just remove `gevent.signal` (even though that's technically a breaking change…the migration path is easy, and without the dynamic aspects even static linters ought to be able to catch any breakage).

**Comment by jamadden** [comment]
Closed in #1530.

**Comment by ulope** [comment]
Thanks!
